scripts/js/queries.js

```diff
@@ -40,39 +42,60 @@ function getDnssecConfig() {
   });
 }
 
+// Fetch database info (earliest timestamp, sizes, ...) from the API and
+// initialize related globals.
+function getDatabaseInfo() {
+  $.getJSON(document.body.dataset.apiurl + "/info/database", data => {
+    // earliest_timestamp is provided in seconds since epoch
+    beginningOfTime = Number(data.earliest_timestamp_disk);
+    // Round down to nearest 5-minute segment (300 seconds)
+    beginningOfTime = Math.floor(beginningOfTime / 300) * 300;
+
+    // endOfTime should be the start of tomorrow in seconds since epoch
+    // We don't use 23:59:59 as the picker increments are set to 5 minutes
+    endOfTime = luxon.DateTime.now().plus({ days: 1 }).startOf("day").toSeconds();
+
+    // If from/until were not provided via GET, default them
+    from ??= beginningOfTime;
+    until ??= endOfTime;
+
+    initDateRangePicker();
+  });
+}
+
 function initDateRangePicker() {
+  const now = luxon.DateTime.now();
+  const minDateDt = luxon.DateTime.fromSeconds(beginningOfTime);
+  const maxDateDt = luxon.DateTime.fromSeconds(endOfTime);
+
   $("#querytime").daterangepicker(
     {
       timePicker: true,
       timePickerIncrement: 5,
       timePicker24Hour: true,
       locale: { format: dateformat },
```

Comment:
If we want to keep using "Sunday" as the first day, we need to add `firstDay` option:

```suggestion
      locale: { 
        format: dateformat,
        firstDay: 7,
      },
```